climada/engine/uncertainty_quantification/calc_cost_benefit.py

```diff
@@ -0,0 +1,285 @@
+        Names of the cost benefit output metris
+        ('tot_climate_risk', 'benefit', 'cost_ben_ratio',
+         'imp_meas_present', 'imp_meas_future')
+    value_unit : str
+        Unit of the exposures value
+    input_var_names : tuple(str)
+        Names of the required uncertainty variables
+        ('haz_input_var', 'ent_input_var', 'haz_fut_input_var', 'ent_fut_input_var')
+    haz_input_var : climada.engine.uncertainty.input_var.InputVar
+        Present Hazard uncertainty variable
+    ent_input_var : climada.engine.uncertainty.input_var.InputVar
+        Present Entity uncertainty variable
+    haz_unc_fut_Var: climada.engine.uncertainty.input_var.InputVar
+        Future Hazard uncertainty variable
+    ent_fut_input_var : climada.engine.uncertainty.input_var.InputVar
+        Future Entity uncertainty variable
+
+    """
+
+    def __init__(self, haz_input_var, ent_input_var,
+                 haz_fut_input_var=None, ent_fut_input_var=None):
+        """Initialize UncCalcCostBenefit
+
+        Sets the uncertainty input variables, the cost benefit metric_names,
+        and the units.
+
+        Parameters
+        ----------
+        haz_input_var : climada.engine.uncertainty.input_var.InputVar
+                        or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard for the present Hazard
+            in climada.engine.CostBenefit.calc
+        ent_input_var : climada.engine.uncertainty.input_var.InputVar
+                        or climada.entity.Entity
+            Entity uncertainty variable or Entity for the present Entity
+            in climada.engine.CostBenefit.calc
+        haz_fut_input_var: climada.engine.uncertainty.input_var.InputVar
+                           or climada.hazard.Hazard, optional
+            Hazard uncertainty variable or Hazard for the future Hazard
+            The Default is None.
+        ent_fut_input_var : climada.engine.uncertainty.input_var.InputVar
+                            or climada.entity.Entity, optional
+            Entity uncertainty variable or Entity for the future Entity
+            in climada.engine.CostBenefit.calc
+
+        """
+
+        Calc.__init__(self)
+        self.input_var_names = ('haz_input_var', 'ent_input_var',
+                             'haz_fut_input_var', 'ent_fut_input_var')
+        self.haz_input_var = InputVar.var_to_inputvar(haz_input_var)
+        self.ent_input_var = InputVar.var_to_inputvar(ent_input_var)
+        self.haz_fut_input_var = InputVar.var_to_inputvar(haz_fut_input_var)
+        self.ent_fut_input_var = InputVar.var_to_inputvar(ent_fut_input_var)
+        self.metric_names = ('tot_climate_risk', 'benefit',
+                             'cost_ben_ratio',
+                             'imp_meas_present', 'imp_meas_future')
+        self.value_unit = self.ent_input_var.evaluate().exposures.value_unit
+
+
+
+    def uncertainty(self, unc_data, pool=None, **cost_benefit_kwargs):
+        """
```

Comment:
I'd prefer this method to return its own `UncOutput` extension, `UncCostBenefitOutput`. Please see my comment at the bottom of unc_ouput.